probe-rs-cli-util/src/argument_handling.rs

```diff
@@ -9,7 +9,7 @@
 ///     "bar=", // Can be "--bar=value" and "--bar value"
 /// ];
 /// ```
-pub fn remove_arguments(arguments_to_remove: &[&'static str], arguments: &mut Vec<String>) {
+pub fn remove_arguments(arguments_to_remove: &Vec<String>, arguments: &mut Vec<String>) {
```

Comment:
Using `&Vec<String>` looks strange to me, I think this should be `&[String]`, because it's not modified.

With the suggested changes to `common_argument`, it should even be possible to use `&[&str]`.

Reply:
Another nice solution would be:

```suggestion
pub fn remove_arguments<T: AsRef<str>>(arguments_to_remove: &[T], arguments: &mut Vec<String>) {
```

With this both `&[&str]`, as well as `&[String]` would be accepted, but we then have to use `as_ref()` on each element of `arguments_to_remove`.